lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -231,50 +267,77 @@ export const migrateGranulesAndFiles = async (
     filesResult: fileMigrationResult,
   };
 
-  let extraSearchParams = {};
+  let extraQueryParams = {};
   type searchType = 'scan' | 'query';
   let dynamoSearchType: searchType = 'scan';
 
-  if (granuleSearchParams.granuleId) {
+  if (granuleMigrationParams.granuleId) {
     dynamoSearchType = 'query';
-    extraSearchParams = {
+    extraQueryParams = {
       KeyConditionExpression: 'granuleId = :granuleId',
       ExpressionAttributeValues: {
-        ':granuleId': granuleSearchParams.granuleId,
+        ':granuleId': granuleMigrationParams.granuleId,
       },
     };
-  } else if (granuleSearchParams.collectionId) {
+  } else if (granuleMigrationParams.collectionId) {
```

Comment:
Should we consider doing a parallel scan/filter on the secondary index here for large collections?       It's probably a premature optimization (... assuming it is an optimization).... so just warrants a ticket, if it's even a good thought.

Reply:
I think we're running out of time for additional tickets, so if it's worth doing, then I can just do it in this ticket. I guess it depends on how many granules you may for a collection, but I suppose that could easily be a very large number. So maybe it is worth doing?

Reply:
I'm not certain the optimization has to be done before release, as the option works, if with less speed.  It'll take time/review... so that's probably a team question.  I tend to think we should ,but I lack context.

Reply:
Just thinking about this more, a parallel scan is an optimization of a scan operation, whereas searching by collectionId is a `query` operation.

It looks like you can supply an index name parameter to `scan` and then you could have a `FilterExpression` to limit the returned results of the scan, but that would only be applied **after the scan**, it would not affect how many results are **evaluated by the scan**. It might be a net performance benefit vs a sequential migration, but it really depends on how many records we're talking about

Reply:
Right, there may be no benefit to `collectionId-granuleId-index` over the main table in that instance, and the collection would need to be a significant portion of the table for it to realize gains.

Given the time constraints, and the improvements in the full migration I think this is fine to punt to an optional  post-release optimization evaluation if we want to take it up.